OpenAI Trossen (#1)
TrossenRobotics/openpi: examples/trossen_ai/README.md

```bash
XLA_PYTHON_CLIENT_MEM_FRACTION=0.9 uv run scripts/train.py <your_custom_config_name> --exp-name=my_experiment --overwrite
```

This allows you to tailor the training process to your dataset and requirements.

Here is an example configuration for training on the Trossen AI dataset:

```python
TrainConfig(
        name="pi0_trossen_transfer_block",
        model=pi0.Pi0Config(paligemma_variant="gemma_2b_lora", action_expert_variant="gemma_300m_lora"),
        data=LeRobotAlohaDataConfig(
            use_delta_joint_actions=False,
            adapt_to_pi=False,
            repo_id="TrossenRoboticsCommunity/bimanual-widowxai-handover-cube",
            assets=AssetsConfig(
                assets_dir="gs://openpi-assets/checkpoints/pi0_base/assets",
                asset_id="trossen",
            ),
            default_prompt="grab and handover the red cube",
            repack_transforms=_transforms.Group(
                inputs=[
                    _transforms.RepackTransform(
                        {

>                             "images": {

Reviewer: Should we have all four cameras here?

                                "cam_high": "observation.images.top",
                                "cam_left_wrist": "observation.images.left",
                                "cam_right_wrist": "observation.images.right",
                            },
                            "state": "observation.state",
                            "actions": "action",
                        }
                    )
                ]
            ),
        ),
        weight_loader=weight_loaders.CheckpointWeightLoader("gs://openpi-assets/checkpoints/pi0_base/params"),
        num_train_steps=20_000,
        batch_size=2,
        freeze_filter=pi0.Pi0Config(
            paligemma_variant="gemma_2b_lora", action_expert_variant="gemma_300m_lora"
        ).get_freeze_filter(),
        # Turn off EMA for LoRA finetuning.
        ema_decay=None,
    ),
```

> We trained on a RTX5090 and fine tuned using LoRA

Reviewer: ```suggestion
We have successfully trained models on an RTX5090 and fine-tuned using LoRA
```

- [OpenPi Fine-Tuned Checkpoint on Hugging Face](https://huggingface.co/shantanu-tr/open_pi_finetune_checkpoint)

After extraction, you can reference this checkpoint when starting the policy server.

## Running Inference with Your Trained Policy

Once training is complete and your checkpoint is ready, you can start the policy server and run the client to perform autonomous tasks.

### Start the Policy Server

This command serves the trained policy, making it available for inference.

Launch the policy server using your trained checkpoint and configuration:

Make sure to run this from project root. This allows us to use LeRobot V0.1.0
```bash
uv run scripts/serve_policy.py policy:checkpoint \
    --policy.config=pi0_trossen_transfer_block \
    --policy.dir=checkpoints/pi0_trossen_transfer_block/test_pi0_finetuning/19999
```

```bash
uv run scripts/serve_policy.py policy:checkpoint \
    --policy.config=pi0_trossen_transfer_block \
    --policy.dir=checkpoints/pi0_trossen_transfer_block/block_transfer_training_100k/99999

> ```

Reviewer: Why two examples here?

Author: That was just me saving the command for myself.

This command serves the trained policy, making it available for inference.

### Start the Client

> Before starting the client we need to built the `LeRobot V0.3.2` package it needs.

Reviewer: ```suggestion
Before starting the client we need to build the `LeRobot V0.3.2` package it needs.
```

> Before starting the client we need to built the `LeRobot V0.3.2` package it needs.

Reviewer: Add a bit at the top about how users need to use different versions of lerobot for different tasks. This is mentioned in different places in this readme, but a note at the top might be better

```bash
cd examples/trossen_ai
GIT_LFS_SKIP_SMUDGE=1 uv sync
GIT_LFS_SKIP_SMUDGE=1 uv pip install -e .
```

> Run the client to interact with the policy server and execute tasks autonomously. Specify the desired task prompt:

Reviewer: ```suggestion
Run the client to interact with the policy server and execute tasks autonomously.
```

> We will use a the `examples/trossen_ai` as root directory for running the client. This is required as the client uses a different version of lerobot (0.3.2) than the training environment.

Reviewer: > lerobot (0.3.2)

You are using inconsistent naming schemes for mentioning lerobot and its version. Pick one and adhere to this scheme throughout

```bash
cd examples/trossen_ai
uv run main.py --mode autonomous --task_prompt "grab red cube"
```

The client will connect to the policy server and perform the specified task using the trained model.

> You can change the cameras and arm ip address in the script by editing

Reviewer: Note where they can find this config
